Sources/Holocron/StorageProvider.swift

```diff
@@ -0,0 +1,55 @@
+//
+//  StorageProvider.swift
+//  Holocron-iOS
+//
+//  Created by Pranjal Satija on 4/10/19.
+//  Copyright © 2019 CocoaPods. All rights reserved.
+//
+
+// MARK: Async Utilities
+/// Performs an (optionally) throwing synchronous operation on an arbitrary background queue
+/// and calls a completion block on the main queue once complete.
+/// - parameter block: A block containing the operation to perform.
+/// - parameter completion: The completion block to call when the operation is finished.
+private func performAsynchronously<T>(_ block: @escaping () throws -> T, completion: @escaping (Result<T, Error>) -> ()) {
+    OperationQueue().addOperation {
```

Comment:
It's been a long day, so someone correct me - but depending on the length of this async operation, we could risk deallocating the `OperationQueue` here, could we not?

Reply:
Nope! The [docs for `OperationQueue`](https://developer.apple.com/documentation/foundation/operationqueue) say:
> Operation queues retain operations until they're finished, and queues themselves are retained until all operations are finished. Suspending an operation queue with operations that aren't finished can result in a memory leak.